app/views/course/assessment/assessments/index.json.jbuilder

```diff
@@ -1,28 +1,130 @@
 # frozen_string_literal: true
-json.array! @assessments do |assessment|
+achievements_enabled = !current_component_host[:course_achievements_component].nil?
+submissions_hash = @assessments.to_h { |assessment| [assessment.id, assessment.submissions] }
+
+json.display do
+  json.student current_course_user&.student? || false
+  json.gamified current_course.gamified?
+  json.randomization current_course.allow_randomization
+  json.achievements achievements_enabled
+  json.bonusAttributes show_bonus_attributes?
+  json.endTimes show_end_at?
+  json.canCreateAssessments can?(:create, Course::Assessment.new(tab: @tab))
+
+  json.category do
+    json.id @category.id
+    json.title @category.title
+    json.tabs @category.tabs.each do |tab|
+      json.id tab.id
+      json.title tab.title
+    end
+  end
+
+  json.tabId @tab.id
+end
+
+json.assessments @assessments do |assessment|
   json.id assessment.id
   json.title assessment.title
-  json.description assessment.description
-  json.published assessment.published
-  json.baseExp assessment.base_exp
-  json.timeBonusExp assessment.time_bonus_exp
-  json.hasPersonalTimes assessment.has_personal_times
-  json.affectsPersonalTimes assessment.affects_personal_times
-  json.tabId assessment.tab_id
-  json.autograded assessment.autograded
-  json.sessionPassword assessment.session_password
-  json.tabbedView assessment.tabbed_view
-  json.skippable assessment.skippable
-  json.delayedGradePublication assessment.delayed_grade_publication
-  json.showPrivate assessment.show_private
-  json.showEvaluation assessment.show_evaluation
-  json.viewPassword assessment.view_password
-  json.usePublic assessment.use_public
-  json.usePrivate assessment.use_private
-  json.useEvaluation assessment.use_evaluation
-  json.allowPartialSubmission assessment.allow_partial_submission
-  json.randomization assessment.randomization
-  json.showMcqAnswer assessment.show_mcq_answer
-  json.showMcqMrqSolution assessment.show_mcq_mrq_solution
-  json.blockStudentViewingAfterSubmitted assessment.block_student_viewing_after_submitted
+
+  json.passwordProtected !assessment.view_password.nil?
+  json.published assessment.published?
+  json.autograded assessment.autograded?
+  json.hasPersonalTimes current_course.show_personalized_timeline_features && assessment.has_personal_times?
```

Comment:
Good catch! Could you help me to add `current_course.show_personalized_timeline_features` check to both `hasPersonalTimes` and `affectsPersonalTimes` in `_video_list_data` too? Thanks!

Reply:
You got it!